lib/features/weather/screen/weather_screen.dart

```diff
@@ -4,39 +4,55 @@ import 'package:flutter_training/features/weather/viewmodel/weather_viewmodel.da
 import 'package:go_router/go_router.dart';
 import 'package:hooks_riverpod/hooks_riverpod.dart';
 
-class WeatherScreen extends ConsumerWidget {
+class WeatherScreen extends HookConsumerWidget {
   const WeatherScreen({super.key});
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
     final state = ref.watch(weatherViewModelProvider);
+    ref.listen(weatherViewModelProvider.select((value) => value.errorOrNull),
+        (_, error) async {
+      if (error == null) {
+        return;
+      }
+      await showErrorDialog(
+        context: context,
+        title: 'エラーが発生しました',
+        message: error.toString(),
+      );
+      // 状態をリセットする
+      ref.read(weatherViewModelProvider.notifier).build();
```

Comment:
[nits]
リセット用のメソッドが用意されているので、そちらを利用していた方が良さそうです。
`ref.refresh` もありますが、今回はそこまでの即時性は求めなくてもいい気がするので、 `ref.invalidate` で十分かなと思います。

https://pub.dev/documentation/flutter_riverpod/latest/flutter_riverpod/WidgetRef/invalidate.html

```suggestion
      ref.invalidate(weatherViewModelProvider);
```

Reply:
ありがとうございました
変更しました
689ab99e5fe6a13f7fac9f2ab29ed2b58d7ba6c1